From 177a5ff52d172488242b3906c0a0ca8d3d8b224c Mon Sep 17 00:00:00 2001
From: Sanjay R Mehta <sanju.mehta@amd.com>
Date: Mon, 28 May 2018 12:05:01 +0530
Subject: [PATCH 4830/5855] Revert "iommu/amd: Limit the IOVA page range to the
 specified addresses"

This reverts commit d7c785b3b8db45a64dad03fa16656e383ba27af6.
---
 drivers/iommu/amd_iommu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/amd_iommu.c b/drivers/iommu/amd_iommu.c
index c7a1d6d..11a13b5 100644
--- a/drivers/iommu/amd_iommu.c
+++ b/drivers/iommu/amd_iommu.c
@@ -3210,7 +3210,7 @@ static void amd_iommu_apply_dm_region(struct device *dev,
 	unsigned long start, end;
 
 	start = IOVA_PFN(region->start);
-	end   = IOVA_PFN(region->start + region->length - 1);
+	end   = IOVA_PFN(region->start + region->length);
 
 	WARN_ON_ONCE(reserve_iova(&dma_dom->iovad, start, end) == NULL);
 }
-- 
2.7.4

